src/renderers/dom/shared/inputValueTracking.js

```diff
@@ -113,40 +113,46 @@ var inputValueTracking = {
     return getTracker(ReactDOMComponentTree.getInstanceFromNode(node));
   },
 
-  trackNode: function(node: ElementWithWrapperState) {
-    if (node._wrapperState.valueTracker) {
+  trackNode(node: ElementWithWrapperState) {
+    if (getTracker(node)) {
       return;
     }
     node._wrapperState.valueTracker = trackValueOnNode(node, node);
   },
 
-  track: function(inst: InstanceWithWrapperState) {
+  track(inst: InstanceWithWrapperState) {
     if (getTracker(inst)) {
       return;
     }
     var node = ReactDOMComponentTree.getNodeFromInstance(inst);
-    attachTracker(inst, trackValueOnNode(node, inst));
+    inst._wrapperState.valueTracker = trackValueOnNode(node, inst);
   },
 
-  updateValueIfChanged(inst: InstanceWithWrapperState | Fiber) {
-    if (!inst) {
+  updateValueIfChanged(subject: SubjectWithWrapperState | Fiber) {
+    if (!subject) {
       return false;
     }
-    var tracker = getTracker(inst);
+    var tracker = getTracker(subject);
 
     if (!tracker) {
-      if (typeof (inst: any).tag === 'number') {
-        inputValueTracking.trackNode((inst: any).stateNode);
+      if (typeof (subject: any).tag === 'number') {
+        inputValueTracking.trackNode((subject: any).stateNode);
       } else {
-        inputValueTracking.track((inst: any));
+        inputValueTracking.track((subject: any));
       }
       return true;
     }
 
     var lastValue = tracker.getValue();
-    var nextValue = getValueFromNode(
-      ReactDOMComponentTree.getNodeFromInstance(inst),
-    );
+
+    var node = subject;
+
+    // TODO: remove check when the Stack renderer is retired
+    if ((subject: any).nodeType !== ELEMENT_NODE) {
+      node = ReactDOMComponentTree.getNodeFromInstance(subject);
```

Comment:
more in the comment above, but the check is purely to avoid calling `getNodeFromInstance` when `inst` is already a Node, as in the Fiber case. The only reason it was added was to handle both renderers, You can see in backport PR that this file isn't even touched (i'm guessing that one probably doesn't suffer this bug btw)